app/src/main/java/com/example/hellobuttons/CounterActivity.kt

```diff
@@ -0,0 +1,27 @@
+package com.example.hellobuttons
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.TextView
+
+class CounterActivity : AppCompatActivity() {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_counter)
+
+        // Get the Intent that started this activity and extract the value
+
+        // option 1
+        val bundle: Bundle? = intent.extras      // built in?
+        val count = bundle?.getInt(COUNTER_VALUE)
+
+        // option 2
+        //val count = intent.getIntExtra(COUNTER_VALUE)
```

Comment:
saw this way of doing it, if you just have one extra, would you do this or always go with a bundle?

Reply:
getIntExtra is just a helper function that accesses the bundle the same way you were above. No matter how many params you have, you can do either one.